pandas/core/reshape/merge.py

```diff
@@ -1705,6 +1706,8 @@ def flip(xs):
             left_values = left_values.view("i8")
             right_values = right_values.view("i8")
             if tolerance is not None:
+                if not isinstance(self.tolerance, Timedelta):
```

Comment:
Don't think we need this `if` statement. (it's okay to pass a `Timedelta` into `Timedelta` I think).